### PR TITLE
Search products page

### PR DESCRIPTION
**Pull request by julioandre**
Completed Search Products Page

### CODE REVIEW

File: src/components/HeroSearchProducts/SearchProducts.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import SearchBar from '../SearchBar';
+import NavBar from '../NavBar';
+const SearchProducts: React.FC = () => {
+  const classes = useStyles();
+
+  return (
+    <>
+      <div className={classes.root}>
+        <CssBaseline />
+        <AppBar position="fixed" className={classes.appBar}>
```
Comment (mutafow) [inline src/components/HeroSearchProducts/SearchProducts.tsx:18]: We already have a NavBar component, you can use that

File: src/components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ const NavBar: React.FC = () => {
   const texts = useTexts();
 
   return (
-    <AppBar position="static">
+    <AppBar position="fixed">
```
Comment (mutafow) [inline src/components/NavBar/NavBar.tsx:9]: Broke the homepage; Don't make such changes without verifying in other components. Besides, I don't think its in the wireframes that we got a fixed navbar 
image

File: src/components/SearchBar/SearchBar.styles.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+
+export const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      padding: '2px 4px',
+      display: 'flex',
+      alignItems: 'center',
+      width: 400,
```
Comment (mutafow) [inline src/components/SearchBar/SearchBar.styles.ts:9]: usually when making generic components its best to not set width, unless they have to have width (like checkbox, image, etc). Later the component adapts its width to the parent element and that makes it a lot more versatile. Now its just the layout so we don't care much but a tip for the future

File: src/components/SideBar/SideBar.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  Checkbox,
+} from '@material-ui/core';
+import { useStyles } from './SideBar.styles';
+import useTexts from '../../hooks/useTexts';
+
+const SideBar: React.FC = () => {
+  const texts = useTexts();
+  const classes = useStyles();
+  const [value, setValue] = React.useState('subcategory');
+  const [state, setState] = React.useState({
```
Comment (mutafow) [inline src/components/SideBar/SideBar.tsx:19]: state, setState => what state? Try to use more descriptive names (e.g checkboxState, setCheckboxState).

File: src/components/SideBar/SideBar.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  FormGroup,
+  Checkbox,
+} from '@material-ui/core';
+import { useStyles } from './SideBar.styles';
+import useTexts from '../../hooks/useTexts';
+
+const SideBar: React.FC = () => {
+  const texts = useTexts();
+  const classes = useStyles();
+  const [value, setValue] = React.useState('subcategory');
```
Comment (mutafow) [inline src/components/SideBar/SideBar.tsx:18]: Also here

File: src/components/SideBar/SideBar.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      </FormControl>
+      <Divider />
+      <FormControl component="fieldset">
+        <FormLabel component="legend">{texts.sideBarSubtitle}</FormLabel>
+        <FormGroup>
+          <FormControlLabel
+            control={<Checkbox checked={premium} onChange={handleChanges} name="premium" />}
+            label="Premium"
+          />
+          <FormControlLabel control={<Checkbox checked={paid} onChange={handleChanges} name="paid" />} label="Paid" />
```
Comment (mutafow) [inline src/components/SideBar/SideBar.tsx:51]: A lot of repetitive code - can be extracted into some sort of .map function. Currently its way too much work to add a new checkbox.

File: src/routes/Router.tsx [hunk 7]
```diff
@@ -2,12 +2,14 @@ import React from 'react';
...
 import Homepage from '../pages/Homepage';
 import Forcompanypage from '../pages/Forcompanypage';
+import SearchProductsPage from '../pages/SearchProductsPage';
 
 const Router: React.FC = () => {
   return (
     <BrowserRouter basename={process.env.PUBLIC_URL}>
       <Route exact path="/" component={Homepage} />
       <Route exact path="/company" component={Forcompanypage} />
+      <Route exact path="/searchproducts" component={SearchProductsPage} />
```
Comment (mutafow) [inline src/routes/Router.tsx:12]: just /products will be a better name

File: src/utils/defaultTexts.ts [hunk 8]
```diff
@@ -22,6 +22,16 @@ const defaultTexts = {
 
   //OptionList.tsx
   optionListTitle: 'Company Options',
+
+  //ProductShowcaseGrid
+  productSGTitle: 'Product',
```
Comment (mutafow) [inline src/utils/defaultTexts.ts:27]: SG means what? More descriptive names make the hell of a difference down the road. Especially when you write this key once and then intelliSense takes over

**Review by mutafow — CHANGES_REQUESTED** [review]

File: src/components/HeroSearchProducts/SearchProducts.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+import { arrayIndexingWithLength } from '../../utils/arrayIndexingWithLength';
+import ProductShowcaseGrid from '../ProductShowcaseGrid';
+import Grid from '@material-ui/core/Grid';
+import { useStyles } from './SearchProducts.styles';
+import { AppBar, CssBaseline, Drawer } from '@material-ui/core';
+
+import SideBar from '../SideBar';
+import SearchBar from '../SearchBar';
+import NavBar from '../NavBar';
+const SearchProducts: React.FC = () => {
```
Comment (mutafow) [inline src/components/HeroSearchProducts/SearchProducts.tsx:11]: The folder HeroSearchProducts/ and the file SearchProducts should be the same name

File: src/components/HeroSearchProducts/SearchProducts.tsx [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+        <Drawer
+          className={classes.drawer}
+          variant="permanent"
+          classes={{
+            paper: classes.drawerPaper,
+          }}
+        >
+          <div className={classes.drawerContainer}>
+            <SideBar />
```
Comment (mutafow) [inline src/components/HeroSearchProducts/SearchProducts.tsx:30]: I think the sidebar should be in a container with the products, see the wireframes, its not all the way to the left.

Also take a look at: Material-ui Box, Container, grid, gridList, Hidden to make the layouts more easily adaptive to responsiveness https://material-ui.com/components/box/

Comment (julioandre) [reply src/components/HeroSearchProducts/SearchProducts.tsx:30]: There is a style property that puts it in the a container

**Review by mutafow — APPROVED** [review]

**Review by mutafow — APPROVED** [review]

**Pull request merged**